Add question on unit testing (#185)
30-seconds/30-seconds-of-interviews: questions/unit-testing.md

### What is Unit Testing ?

### Answer

> In layman terms, unit Testing is a level of software testing where individual units or components of a software are tested independently.

Reviewer: ```suggestion
Unit Testing is a level of software testing where individual units or components of software are tested independently.
```

> Generally, we write code tests using frameworks like Mocha and Chai. We validate the module by giving it all types of inputs into it, and expect the output we are looking for.

Reviewer: ```suggestion
Tests are written using frameworks like Mocha, Chai, and Jest. The software is validated by giving it different inputs, and checking to see that the desired output is returned.
```

> **Benefits**:

Reviewer: This section should be in Good to hear

Author: The Benefits ?

Reviewer: Yup. You wouldn't mention them as part of the answer, but it would be good to hear and even good to research on your own.

Author: Done ! @skatcat31

- Keeps the unit working.
- The unit is always executed on any new change.

### Good to hear

> Take a look at different testing frameworks like Jest which are way easier to use.

Reviewer: Edited to mention Jest above.

> 

Reviewer: ```suggestion
Test the software in a way the end user would be expected to use it.

```

### What is Unit Testing ?

### Answer

Unit Testing is a level of software testing where individual units or components of software are tested independently.

Tests are written using frameworks like Mocha, Chai, and Jest. The software is validated by giving it different inputs, and checking to see that the desired output is returned.

> ### Good to hear

Reviewer: The good to hear section needs some reformatting to look like other submissions

> **Benefits**:

Reviewer: you can remove the benefits bold here entirely now, since the benefits are the good to hear
